### PR TITLE
[stable/prometheus-redis-exporter]Add redis authoration configuration in chart

### PR DESCRIPTION
**Pull request by aixeshunter**
Signed-off-by: aixeshunter <aixeshunter@gmail.com>

no

#### What this PR does / why we need it:
There is no redis password configuration in prometheus-redis-exporter chart.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by aixeshunter** [comment]
@mattfarina, @acondrat  Please review and approve this PR, thx. :)

File: stable/prometheus-redis-exporter/values.yaml [hunk 1]
```diff
@@ -58,3 +58,13 @@ serviceMonitor:
 # script:
 #   configmap: prometheus-redis-exporter-script
 #   keyname: script
+
+auth:
+  # when redis use password to attach
```
Comment (acondrat) [inline stable/prometheus-redis-exporter/values.yaml]:
```suggestion
  # Use password authentication
```

File: stable/prometheus-redis-exporter/values.yaml [hunk 2]
```diff
@@ -58,3 +58,13 @@ serviceMonitor:
...
+  enabled: false
+  # redis password stored in Secret
```
Comment (acondrat) [inline stable/prometheus-redis-exporter/values.yaml]:
```suggestion
  # Use existing secret
```

File: stable/prometheus-redis-exporter/values.yaml [hunk 3]
```diff
@@ -58,3 +58,13 @@ serviceMonitor:
...
+  secret:
+    name: ""
+    key: ""
+  # when redis password not be stored in Secret, point it directly
```
Comment (acondrat) [inline stable/prometheus-redis-exporter/values.yaml]:
```suggestion
  # Redis password (when not stored in a secret)
```

**Review by acondrat — CHANGES_REQUESTED** [review]
Please document all new values in the readme. Also, please make sure to document precedence when both secret and plain passwords are defined.

**Comment by aixeshunter** [comment]
> Please document all new values in the readme. Also, please make sure to document precedence when both secret and plain passwords are defined.

@acondrat Thx for review, I resolved all. Please check it again.

File: stable/prometheus-redis-exporter/README.md [hunk 4]
```diff
@@ -69,6 +69,10 @@ The following table lists the configurable parameters and their default values.
 | `serviceMonitor.targetLabels`  | Set of labels to transfer on the Kubernetes Service onto the target.  |             |
 | `script.configmap`     | Let you run a custom lua script from a configmap. The corresponding environment variable `REDIS_EXPORTER_SCRIPT` will be set automatically ||
 | `script.keyname`       | Name of the key inside configmap which contains your script ||
+| `auth.enabled`       | Specifies whether redis use authentication | `false` |
+| `auth.secret.name`       | Name of redis existing secret (ignores redisPassword) ||
```
Comment (acondrat) [inline stable/prometheus-redis-exporter/README.md]:
```suggestion
| `auth.secret.name`       | Name of existing redis secret (ignores redisPassword) ||
```

File: stable/prometheus-redis-exporter/README.md [hunk 5]
```diff
@@ -69,6 +69,10 @@ The following table lists the configurable parameters and their default values.
 | `serviceMonitor.targetLabels`  | Set of labels to transfer on the Kubernetes Service onto the target.  |             |
 | `script.configmap`     | Let you run a custom lua script from a configmap. The corresponding environment variable `REDIS_EXPORTER_SCRIPT` will be set automatically ||
 | `script.keyname`       | Name of the key inside configmap which contains your script ||
+| `auth.enabled`       | Specifies whether redis use authentication | `false` |
```
Comment (acondrat) [inline stable/prometheus-redis-exporter/README.md]:
```suggestion
| `auth.enabled`       | Specifies whether redis uses authentication | `false` |
```

File: stable/prometheus-redis-exporter/README.md [hunk 6]
```diff
@@ -69,6 +69,10 @@ The following table lists the configurable parameters and their default values.
...
+| `auth.secret.name`       | Name of redis existing secret (ignores redisPassword) ||
+| `auth.secret.key`       | Key of redis existing secret ||
```
Comment (acondrat) [inline stable/prometheus-redis-exporter/README.md]:
```suggestion
| `auth.secret.key`       | Name of key containing password to be retrieved from the existing secret ||
```

File: stable/prometheus-redis-exporter/README.md [hunk 7]
```diff
@@ -69,6 +69,10 @@ The following table lists the configurable parameters and their default values.
...
+| `auth.redisPassword`       | Point the redis password when not stored in a secret ||
```
Comment (acondrat) [inline stable/prometheus-redis-exporter/README.md]:
```suggestion
| `auth.redisPassword`       | Redis password (when not stored in a secret) ||
```

**Review by acondrat — CHANGES_REQUESTED** [review]
Please review values' description. 
Sorry for delays on this review.

**Review by acondrat — APPROVED** [review]
/lgtm

**Comment by eh-am** [comment]
```
key: {{ .Values.secret.key }}
```

shouldn't it be `{{ .Values.auth.secret.key }}`?

**Comment by acondrat** [comment]
> ```
> key: {{ .Values.secret.key }}
> ```
> 
> shouldn't it be `{{ .Values.auth.secret.key }}`?

Yes, I think so. Good catch!

/lgtm cancel

**Pull request merged**